### PR TITLE
Added google text to speech as alternative ( cheaper )

### PR DESCRIPTION
**Pull request by btimmy**
### Background

No discussions, but I was looking for a cheaper alternative to elevenlabs and saw that google's text to speech is much cheaper. Not quite as good as eleven, but way better than standard tts, and the free quota on elevenlabs ran out pretty quickly. Google's TTS is widely used and has good, natural sounding voices for a much cheaper cost.

### Changes
I added a "google_speak.py" file which inherits the base class just like the other voice classes that are already written, which means it'll nicely integrate with the current code base. The only thing the user needs to do is to get credentials from google and specify the path to the credentials-file in the .env.template file). If a path is set, TTS will use google.

### Documentation
I added information to README.md which clearly describes how to use google TTS if the user wants to. I placed the info beneath the eleven labs information:

if you want to use google text to speech instead:
1. Visit cloud.google.com libraries and activate google text to speech API

2. Generate a client_secrets or service_account.json file from "api's and services" -> "credentials", save the generated file on your computer.

3. Specify env variable GOOGLE_SERVICE_ACCOUNT_JSON={path_to_client_secret_or_service_account.json} this should be a path to the credentials file. 

4. Run "python -m autogpt --speak" as normal.

### Test Plan
1. I got the service_account.json file from google, and downloaded it to my computer.
2. I. specified the path to the file using the new .env variable GOOGLE_SERVICE_ACCOUNT_JSON found in .env.template
3. I added the openAI api key to the .env.template file too.
4. I renamed .env.template to .env
5. I created a virtual environment using "virtualenv venv" and activated it with "source venv/bin/activate"
6. I installed all packages (including google's tts that I added to requirements.txt) using "pip install -r requirements.txt"
7. I got an error ( That I also got before doing any changes, which is why I'm not fixing this in this pull request ).
The error was: ModuleNotFoundError: No module named 'AppKit' and it was easily solved for me with pip install pyobjc. I didn't want to include this in the request because it might have to do with my machine specifically and it probably deserves a discussion first.
8. I ran the script "python -m autogpt --speak" and got the soothing voice from google's TTS :)

### PR Quality Checklist
- [ X ] My pull request is atomic and focuses on a single change.
- [ X ] I have thoroughly tested my changes with multiple different prompts.
- [ X ] I have considered potential risks and mitigations for my changes.
- [ X ] I have documented my changes clearly and comprehensively.
- [ X ] I have not snuck in any "extra" small tweaks changes 

Testing requires human quality reassurance (ears), so I've tested everything manually as described above.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -216,6 +216,17 @@ python -m autogpt --speak
...
+### if you want to use google text to speech instead:
+# 1. Visit cloud.google.com libraries and activate google text to speech API
+
+# 2. Generate a client_secrets or service_account.json file from "api's and services" -> "credentials", save the generated file on your computer.
+
+# 3. Specify env variable GOOGLE_SERVICE_ACCOUNT_JSON={path_to_client_secret_or_service_account.json} this should be a path to the credentials file. 
+
+# 4. Run "python -m autogpt --speak" as normal.
+#####
+
```
Comment (nponeccop) [inline README.md:229]: The markdown is wrong. # has a special meaning there

File: .env.template [hunk 2]
```diff
@@ -171,3 +171,9 @@ TW_CONSUMER_KEY=
 TW_CONSUMER_SECRET=
 TW_ACCESS_TOKEN=
 TW_ACCESS_TOKEN_SECRET=
+
+
+##add path to client_secrets.json file or service_account.json file to enable google text to speech.
+GOOGLE_SERVICE_ACCOUNT_JSON=
+GOOGLE_VOICE_1_ID=
+GOOGLE_VOICE_2_ID=
```
Comment (nponeccop) [inline .env.template:179]: Whitespace error at EOL. Fix by pressing Enter once after the = sign

File: .gitignore [hunk 3]
```diff
@@ -22,6 +22,10 @@ logs
 *.log
 *.mp3
 
+#private files
+*service_account.json
+*client_secret.json
+
```
Comment (nponeccop) [inline .gitignore:28]: Doesn't look right

**Review by nponeccop — CHANGES_REQUESTED** [review]

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Comment by Pwuts** [comment]
Closing because no activity since original submission, and it would fit better as a plugin instead of a module in the main codebase

**Pull request closed without merging**